src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -184,13 +236,20 @@ struct nonmaxsuppression
             result.y = {static_cast<double>(start[0]), static_cast<double>(start[2])};
         }
 
+        result.sort();
+
```

Comment:
Line 220: this could take a const ref..?

Reply:
Its just an iterator so its cheap to copy.

Reply:
(const reference won't hurt a templated function.)

Also, compute_nms can take const references to avoid copies. Thanks.

Reply:
> (const reference won't hurt a templated function.)

It can be a pessizimization since it introduces aliasing. 

> Also, compute_nms can take const references to avoid copies.

Those are tensor views so they dont need to be passed as reference either. We also usually never pass them as reference to the visitors.

Reply:
https://www.oreilly.com/library/view/c-coding-standards/0321113586/ch10.html will hopefully convince you, and not be so pessimistic about const references :-)

Reply:
Sure, but pass-by-reference is not appropriate here since you are already passing a pointer type. It much faster to pass a pointer than to pass an address to another pointer.